app/routes/calendar/new.tsx

```diff
@@ -287,15 +284,22 @@ function DescriptionTextarea() {
 
 function DatesInput() {
   const { t } = useTranslation(["common", "calendar"]);
-  const { eventToEdit } = useLoaderData<typeof loader>();
-  const [datesCount, setDatesCount] = React.useState(
-    eventToEdit?.startTimes.length ?? 1
-  );
-  const isMounted = useIsMounted();
 
+  // React hook that keeps contains an array of parameters that corresponds to each DateInput child object generated
+  const [datesInputState, setDatesInputState] = React.useState([
+    {
+      finalDateInputDate: new Date(),
+      index: 0,
+    },
```

Comment:
Mmm, ambiguous naming in the tsx page (though that is partially on default routing mechanism) - I would have never suspected that it would use the same page when editing a pre-existing Calendar due to the name of the file.

Yes, we'll definitely have to handle this use case scenario when editing a pre-existing Calendar

Reply:
> 1. Is the `index` relevant here, or is it enough to know the position inside the array? I feel like currently this would always look like `[{index: 0, ... }, {index: 1, ... }, {index: 2, ... }]` and so on (leaving finalDateInputDate out for brevity). I think we could get rid of this, and maybe even have the state only be of type `Array<Date>` or `Array<Date | null>`? (See later comment for why null)

Initially, I only included this because we are iterating through each object in the inputState's array & thought that we couldn't keep track of the index. I now remember that you can get the index during the iteration with the 2nd arg in `Array.map()`, so I will refactor this now & get rid of the object's index attribute :)

Reply:
#2 is the most important issue to resolve here (just leaving a note for myself here)

Reply:
> 2\. We need to properly initialize the state here.One gotcha is, that the route `new.tsx` is not only used for creating _new_ events, but also _edit_ existing ones. That's why the `datesCount` state got initialized with `eventToEdit?.startTimes.length ?? 1`, meaning: "If we are editing an event, use that event's data as the initial state, otherwise use 1"Currently when editing an existing event, this code will just reset the date inputs, possibly leading to data loss if the user didn't notice. We need to implement the same thing for the new state, it should initialize with the existing dates on the event, if it exists.

Okay, I handled this now too =)

Can you please have another look when you have time? Thanks again for all your feedback!